packages/orchestration/src/workflow/local-workflow.ts

```diff
@@ -50,10 +50,13 @@ export class LocalWorkflow {
     this.handlers = new Map(globalWorkflow.handlers_)
 
     let container
-
     if (!Array.isArray(modulesLoaded) && modulesLoaded) {
       if (!("cradle" in modulesLoaded)) {
-        container = createMedusaContainer(modulesLoaded)
+        container = createMedusaContainer()
+
+        for (const [name, registration] of Object.entries(modulesLoaded)) {
+          container.register(name, asValue(registration))
+        }
```

Comment:
comment: I can't tell you exactly why I needed to add this workaround, but it seems the `awilix.createContainer` call in the `createMedusaContainer` util failed to properly handle the conversion in this specific case:

https://github.com/medusajs/medusa/blob/e61c46c36da51d24e2294ce2b1d5b568bfe9526c/packages/utils/src/common/medusa-container.ts#L43-L56

The one conversion that happens on line 44.

Reply:
Have you tried to use `createContainerLike(modulesLoaded)`?
otherwise we can bring that loop to register all the keys using asValue to the createMedusaContainer helper if the "cradle" is not present in the first arg.

Reply:
I believe if you use the createContainerLike util before calling the createContainer it would work, could you try?

Reply:
I would avoid that loop as we might loose the life time of the object that are registered by doing it like that i think

Reply:
You mean like so:
```
const containerLike = createContainerLike(modulesLoaded)
container = createMedusaContainer(containerLike)
```